README.md

```diff
@@ -1 +1,128 @@
+     - will take the sutable action for each operation to create, update and delete
+     - waits on them and report the state
+  4. For applying the changes we have subi and node pkgs which creates contracts/deployments on the grid
+  5. if a user wishes to apply any changes, they should provide their new state, and their current deployment ids.
+  6. the deployment manager should be responsible of reverting the applied changes if some error happens midway.
+
+- ### **How the deployment manager calculates changes:**
+
+- the deployment manager receives informantion about old and new deployments, then decides which operations need to be performed.
+
+  1. deployments that need to be created are present in the new deployments, and not in the old deployments.
+  2. deployments that need to be deleted are present in the old deployments, and not in the new deployments.
+  3. deployments that need to be updated are present in both old and new deployments, but they must have different hashes.
+
+- ### **Creating a new Deployment:**
+
+  1. deployments must first be signed and validated.
+  2. a deployment contract then must be created.
+  3. Then, the deployment should be deployed on the node.
+  4. If some error happens while trying to deploy on the node, the contract should be canceled to avoid leaking a contract.
+  5. after deployment creating, the function should only return after waiting on all workloads to be StateOK.
+
+- ### **Deleting a deployment:**
+
+  1. Deleting a deployment is only done on the chain, by canceling the deployment contract.
+
+- ### **Updating a deployment:**
+
+  1. a deployment should be updated if and only if hashes differ or a workload name was changed.
+  2. if a deployment should be updated, its version should be incremented.
+  3. if a workload should be updated, its version should match the deployment version.
+  4. if a workload shouldn't be updated, its version statys the same.
+  5. the deployment contract should then, be updated.
+  6. The deployment on the node should be updated.
+  7. after deployment update, the function should only return after waiting on all workloads to be StateOK.
+
+- ### **Generating a versionless deployemnt:**
+
+  1. a user should first generate the appropriate deployment builder using the `builder` package.
+  2. the `builder` package then generates the grid.Deployment objects.
+
+- ### **Reverting a deployment:**
+
+  1. before applying any change, the deployment manager should first retreive the current state from the nodes `oldState`.
+  2. at first, `currentState` of deployment ids is provided by the user (as `oldDeloymentIDs`).
+  3. every contract deletion or creation, should directly be reflected in the `currentState`.
+  4. if some error happens while applying some change, the deployment manager should revert to its old state using the `currentState` as the `oldDeploymentIDs` and the `oldState` as the `newDeployemnts`.
+
+    Example:
```

Comment:
I believe the example should be moved after the reverting deployment section with multiple examples, just create, update, and delete

then introduce more complex examples